### DISCUSSION TITLE
SQLA Code Map for developers

### DISCUSSION THREAD

**Discussion by jenstroeger** [sqlalchemy/sqlalchemy#11713] [Ideas]
Hello, today I stumbled upon the [Code Visualization project on Github Next](https://githubnext.com/projects/repo-visualization/), and pointed it at the SQLA repo ([link](https://mango-dune-07a8b7110.1.azurestaticapps.net/?repo=sqlalchemy%2Fsqlalchemy)) — quite insightful with interesting [future ideas](https://githubnext.com/projects/repo-visualization/#potential-future-directions).

They also provide a Github Action to generate that data automatically and to integrate it into documentation ([docs](https://githubnext.com/projects/repo-visualization/#integrate-into-your-own-projects)). I thought that might be a useful starting point to add to the SQLA [Develop](https://www.sqlalchemy.org/develop.html) page?

Didum, Jens 🤓

**Comment by CaselIT** [comment]
Hi,

That's an interesting visualization, but I'm not sure about its utility. As far as I can see, it just group together files by folder, without much other information.

**Reply by jenstroeger** [reply]
I agree — at the moment here’s not much to it but their [Future Directions](https://githubnext.com/projects/repo-visualization/#what-files-are-connected) seems promising…

**Reply by CaselIT** [reply]
let's see how it evolves!
